state-reconstruct-fetcher/src/l1_fetcher.rs

```diff
@@ -58,6 +56,32 @@ pub struct L1FetcherOptions {
     pub disable_polling: bool,
 }
 
+#[derive(Clone)]
+struct FetcherCancellationToken(CancellationToken);
+
+impl FetcherCancellationToken {
+    const LONG_TIMEOUT_S: u64 = 120;
+
+    pub fn new() -> FetcherCancellationToken {
+        FetcherCancellationToken(CancellationToken::new())
+    }
+
+    pub async fn cancelled_else_long_timeout(&self) {
```

Comment:
Actually this could be useable for all calls to sleep, if it had the duration parameter - something like `sleep_with_cancel`...

Reply:
I agree, do you mind if that's for a follow-up PR instead?

Reply:
Sure - we can postpone that...